File: doc/_i18n/en/tutorials/introduction/dual-arm-controller/logic_setup.cpp
// In the header
enum ControllerPhase
{
  IDLE = 0,
  STARTED,
  MOVE
};
// A private property of our controller
ControllerPhase phase = IDLE;
// In the run function
bool DualArmController::run()
{
  if(phase_ == IDLE && 0 /** we write this condition later */)
  {
    /** Setup the STARTED phase */
    phase_ = STARTED;
  }
  else if(phase_ == STARTED && 0 /** we write this condition later */)
  {
    /** Setup the MOVE phase */
    phase_ = MOVE;
  }
  else if(phase_ == MOVE && 0 /** we write this condition later */) {}
  return mc_control::MCController::run();
}

File: doc/_i18n/en/tutorials/introduction/dual-arm-controller/logic_setup.py
# Declare constants
class ControllerPhase:
    IDLE = 0
    STARTED = 1
    MOVE = 2


# In constructor
def __init__(self, rm, dt):
    self._phase = IDLE


# In the run callback
def run_callback(self):
    if self._phase == ControllerPhase.IDLE and False:  # We write this condition later
        # Setup the STARTED phase
        self._phase = ControllerPhase.STARTED
    elif (
        self._phase == ControllerPhase.STARTED and False
    ):  # We write this condition later
        # Setup the MOVE phase
        self._phase = ControllerPhase.MOVE
    elif self._phase == ControllerPhase.MOVE and False:  # We write this condition later
        # Continue the MOVE phase
        pass
    return True
